backend/modules/simulation/service.py

```
@@ -148,10 +161,25 @@ async def process_chat_message(
+            if user_progress.sandbox_id and next_scene_id:
+                db_next_scene = self.repository.get_scene_by_id(next_scene_id)
+                if db_next_scene and getattr(db_next_scene, "scene_type", "conversation") == "code_challenge":
+                    scene_data_files = getattr(db_next_scene, "data_files", None)
+                    if scene_data_files:
+                        try:
+                            from common.services.sandbox_service import sandbox_service
+                            count = await sandbox_service.upload_scene_data_files(
+                                user_progress.sandbox_id, scene_data_files
+                            )
+                            logger.info(f"[SERVICE] Uploaded {count} data files for scene {next_scene_id}")
+                        except Exception as e:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)